From 8b3022ae1c491f476980ccdb27d36c3924cab1a7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20D=C3=B6singer?= <stefan@codeweavers.com>
Date: Thu, 4 Jul 2013 22:59:33 +0200
Subject: [PATCH 073/127] wined3d: Get rid of the end_scene flush and finish
Reply-To: wine-devel <wine-devel@winehq.org>

Either keep the patch this way, or send a flush through the CS. I'm not
sure it's worth it though, this has never had any real performance
impact.
---
 dlls/wined3d/device.c | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/dlls/wined3d/device.c b/dlls/wined3d/device.c
index b27c602..3aa31b3 100644
--- a/dlls/wined3d/device.c
+++ b/dlls/wined3d/device.c
@@ -3256,8 +3256,6 @@ HRESULT CDECL wined3d_device_begin_scene(struct wined3d_device *device)
 
 HRESULT CDECL wined3d_device_end_scene(struct wined3d_device *device)
 {
-    struct wined3d_context *context;
-
     TRACE("device %p.\n", device);
 
     if (!device->inScene)
@@ -3266,16 +3264,6 @@ HRESULT CDECL wined3d_device_end_scene(struct wined3d_device *device)
         return WINED3DERR_INVALIDCALL;
     }
 
-    context = context_acquire(device, NULL);
-    /* We only have to do this if we need to read the, swapbuffers performs a flush for us */
-    if (wined3d_settings.cs_multithreaded)
-        context->gl_info->gl_ops.gl.p_glFinish();
-    else
-        context->gl_info->gl_ops.gl.p_glFlush();
-    /* No checkGLcall here to avoid locking the lock just for checking a call that hardly ever
-     * fails. */
-    context_release(context);
-
     device->inScene = FALSE;
     return WINED3D_OK;
 }
-- 
1.8.1.5

